optimum/neuron/models/qwen2/modules.py

```diff
@@ -0,0 +1,83 @@
+# Copyright Amazon Web Services and its Affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from transformers_neuronx import dtypes, module, utils
```

Comment:
couldn't we reuse the modules of llama ?

Reply:
No, because attention is unfortunately not configurable, and the Qwen2Attention is specific because of the biases in the linears. Then it forces us to redefine all layers hierarchy.